### ISSUE TITLE
Full And Sanitized Node Documentation

### ISSUE THREAD

**Issue by stefgodin** [nvim-tree/nvim-tree.lua#2668]
**Is this a question?**
No

**Can this functionality be implemented utilizing API?**
Not a functionality per say.

**Is your feature request related to a problem? Please describe.**
Many functions of the API retrieves nodes such as `get_node_under_cursor()`.
There doesn't seem to be any documentation about the structure of a node except in the [sort API section](https://github.com/nvim-tree/nvim-tree.lua/blob/8cbb1db8e90b62fc56f379992e622e9f919792ce/doc/nvim-tree-lua.txt#L679). But there are some missing fields in the doc.

**Describe the solution you'd like**
It would be nice to have a small section of the documentation dedicated to the content of a node. This would make it officially part of the API.

**Describe alternatives you've considered**
Check the [node builder source code](https://github.com/nvim-tree/nvim-tree.lua/blob/master/lua/nvim-tree/explorer/node-builders.lua) but that does not prevent invisible breaking changes to the node API.

**Additional context**
This issue is raised after a [short post on reddit](https://www.reddit.com/r/neovim/comments/1akjoc3/what_can_i_do_with_a_nvimtree_node/).

I don't know if this qualifies as a feature request, feel free to move it around.

nvim-tree is an awesome plugin and you did a really good job.

**Comment by alex-courtis** [comment]
We do have https://github.com/nvim-tree/nvim-tree.lua/blob/6a99f5af78b5dd531d49ec0a5158dd0f9a7c992b/lua/nvim-tree/node.lua to document all the contents of various nodes, however that's not super informative.

Proposal: add more descriptions and include it in help.

**Comment by alex-courtis** [comment]
@stefgodin _meta api documentation has been added for all the `nvim_tree.api.Node` types, as they are exposed via API.

https://github.com/nvim-tree/nvim-tree.lua/blob/master/lua/nvim-tree/_meta/api.lua

Does that satisfy your requirements?

More work needs to be done to add them to `api.lua` and help, as mentioned in #2832

**Comment by stefgodin** [comment]
@alex-courtis  Yes! Thank you. Feel free to close the issue.

**Comment by alex-courtis** [comment]
Partially addressed by #3244 

Outstanding: `nvim_tree.api.Node` subclasses, with `:as` etc. documented
Requires: conversion of the classes to classic

Most API and callbacks use real nodes, only `get_nodes` and decorators use `nvim_tree.api.*Node`. Decision:
1. change API and callbacks to use `nvim_tree.api.*Node`
2. abandon the sanitized api nodes and just return real nodes, classed as `nvim_tree.api.*Node`

1 may break users, however the actual nodes are not documented API, so it is a reasonable change.
